sign_oca/tests/test_sign_portal.py

```diff
@@ -71,3 +105,103 @@ def test_portal(self):
+            "sign_requests": sign_requests,
+            "pager": {
+                "page_count": 2,
+                "offset": 0,
+                "page": {"url": "/my/sign_requests", "num": 1},
+                "page_first": {"url": "/my/sign_requests", "num": 1},
+                "page_start": {"url": "/my/sign_requests", "num": 1},
+                "page_previous": {"url": "/my/sign_requests", "num": 1},
+                "page_next": {"url": "/my/sign_requests/page/2", "num": 1},
+                "page_end": {"url": "/my/sign_requests/page/2", "num": 1},
+                "page_last": {"url": "/my/sign_requests/page/2", "num": 1},
+                "pages": [{"url": "/my/sign_requests", "num": 1}],
+            },
+            "default_url": "/my/sign_requests",
+        }
+        expected_domain = [
+            ("request_id.state", "=", "sent"),
+            ("partner_id", "child_of", [self.env.user.partner_id.id]),
+            ("signed_on", "=", False),
+        ]
+        expected_home_values = {"sign_oca_count": 1}
+
+        class Capture:
+            mock_response = Mock()
+
+            def get_sign_requests_domain(self, request):
+                self.mock_response.domain = expected_domain
+                return self.mock_response
+
+            def _prepare_sign_portal_rendering_values(
+                self, request, page=1, sign_page=False, **kwargs
+            ):
+                self.mock_response.values = expected_values
+                return self.mock_response
+
+            def _prepare_home_portal_values(self, counters):
+                self.mock_response.home_values = {"sign_oca_count": len(sign_requests)}
+                return self.mock_response
+
+        capture = Capture()
+
+        with patch(
+            "odoo.addons.sign_oca.controllers.main.PortalSign.get_sign_requests_domain",
+            capture.get_sign_requests_domain,
+        ):
+            response = self.portal_sign.get_sign_requests_domain(capture.mock_response)
+            self.assertEqual(response.domain, expected_domain)
+
+        with patch(
+            (
+                "odoo.addons.sign_oca.controllers.main.PortalSign"
+                "._prepare_sign_portal_rendering_values"
+            ),
+            capture._prepare_sign_portal_rendering_values,
+        ):
+            response = self.portal_sign._prepare_sign_portal_rendering_values(
+                request=capture.mock_response, page=1, sign_page=False, kwargs={}
+            )
+            self.assertEqual(response.values, expected_values)
+
+        with patch(
+            (
+                "odoo.addons.sign_oca.controllers.main.PortalSign"
+                "._prepare_home_portal_values"
+            ),
+            capture._prepare_home_portal_values,
+        ):
+            response = self.portal_sign._prepare_home_portal_values(
+                counters=["sign_oca_count"]
+            )
+            self.assertEqual(response.home_values, expected_home_values)
```

Comment:
I think it is not necessary to test all possible methods or use Mock, it would not be enough to do a portal test with a tour.

Reply:
I think this test is not necessary, because it is testing “_not necessary_” things, all this is already being tested with the portal test (_test_sign_requests_) which is defined later.